### DISCUSSION TITLE
Can't seem to get code generation working

### DISCUSSION THREAD

**Discussion by barrykaplan** [riok/mapperly#308] [Q&A] [answered]
I get errors like
```
/home/bkaplan/dev.hbk/reliasoft/reli/src/Reli/Core/FMEA/Smoke.cs(16,5): error RMG008: Could not create mapping from Reli.Core.FMEA.Car to CarDto. Consider implementing the mapping manually. [/home/bkaplan/dev.hbk/reliasoft/reli/src/Reli/Reli.csproj]
/home/bkaplan/dev.hbk/reliasoft/reli/src/Reli/Core/FMEA/Smoke.cs(16,20): error CS0246: The type or namespace name 'CarDto' could not be found (are you missing a using directive or an assembly reference?) [/home/bkaplan/dev.hbk/reliasoft/reli/src/Reli/Reli.csproj]
/home/bkaplan/dev.hbk/reliasoft/reli/src/Reli/Riok.Mapperly/Riok.Mapperly.MapperGenerator/CarMapper.g.cs(6,24): error CS0246: The type or namespace name 'CarDto' could not be found (are you missing a using directive or an assembly reference?) [/home/bkaplan/dev.hbk/reliasoft/reli/src/Reli/Reli.csproj]
```

I can clone Mapperly repo and load the solution into Rider and it works just fine there. I've tried to duplicate all the relevant packages in the project to no avail.

I don't see anything in the docs that mentions even having to include `Microsoft.CodeAnalysis.Analyzers`. Am I missing something?

**Comment by barrykaplan** [comment]
Hmm, now that I look at the samples/, the CarDto class seems to be included -- ie, not generated. Now I'm really confused.

**Comment by barrykaplan** [comment]
But I can run all the tests in the Mapperly project both on the command line and within Rider...

**Comment by barrykaplan** [comment]
Ok, maybe I have misunderstood. I thought that it would also generate a default DTO if one did not exist (and then I could customize the mapping from there). But that is not the case... ?

**Reply by latonz** [reply, marked as answer]
Mapperly does not generate DTO's. It only generates method bodies for the defined mapping methods containing the mapping code from one type to another.
